src/Rings/walk.jl

```diff
@@ -0,0 +1,1724 @@
+				continue
+			end
+		end
+		G = standard_step_without_int32_check(G, w, T)
+		if infoLevel >= 1
+			println(w)
+			if infoLevel == 2
+				println(G)
+			end
+		end
+		currweight = w
+		raise_step_counter()
+	end
+end
+
+###############################################################
+# Standard step without checking of the entries of a given weight vector.
+###############################################################
+
+function standard_step_without_int32_check(
+	G::Oscar.IdealGens,
+	w::Vector{Int},
+	T::Matrix{Int},
+)
+	R = base_ring(G)
+	ordAlt = G.ord
+	ordNew = create_order(R, w, T)
+	Gw = initials(G, w)
+	H = groebner_basis(ideal(Gw), ordering = ordNew, complete_reduction = true, algorithm = :buchberger)
+	#H = liftGW2(G, R, Gw, H, Rn)
+	H = lift(G, ordAlt, H, ordNew)
+	return interreduce_walk(H)
+end
+
+
+#################################################################
+# Procedures of the fractal walk.
+# The fractal walk is proposed by Amrhein & Gloor (1998).
+#################################################################
+
+# lifts the Groebner basis G to the Groebner basis w.r.t. in the Fractal Walk like it´s done in Fukuda et. al (2005).
+function lift_fractal_walk(
+	G::Oscar.IdealGens,
+	H::Oscar.IdealGens,
+	orderingNew::MonomialOrdering,
+)
+	R = base_ring(G)
+	G.isGB = true
+	G = Oscar.IdealGens(
+		R,
+		[
+			gen -
+			Oscar.IdealGens([reduce(gen, gens(G), ordering = G.ord, complete_reduction = true)], H.ord)[1] for
+			gen in gens(H)
+		],
+		orderingNew,
+	)
+	G.isGB = true
+	return G
+end
+
+# returns ´true´ if all polynomials of the given array are monomials.
+function ismonomial(Gw::Vector{T}) where {T <: MPolyRingElem}
+	for g in Gw
+		if length(coefficients(g)) > 1
+			return false
+		end
+	end
+	return true
+end
+
+# returns ´true´ if all polynomials of the given array are binomials or less.
+function isbinomial(Gw::Vector{T}) where {T <: MPolyRingElem}
+	for g in Gw
+		if length(coefficients(g)) > 2
+			return false
+		end
+	end
+	return true
+end
+
+# returns the next t to compute the next weight vector w(t) = w + t * (tw - w) like it´s done in Amrhein & Gloor (1998). This Method is NOT tested sufficiently.
+function nextT(
+	G::Oscar.IdealGens,
+	w::Array{T, 1},
+	tw::Array{K, 1},
+) where {T <: Number, K <: Number}
+	if (w == tw)
+		return [0]
+	end
+	tmin = 2
+	t = 0
+	for g in gens(G)
+		a = Singular.leading_exponent_vector(g)
+		d = Singular.exponent_vectors(tail(g))
+		for v in d
+			frac = (dot(w, a) - dot(w, v) + dot(tw, v) - dot(tw, a))
+			if frac != 0
+				t = (dot(w, a) - dot(w, v)) // frac
+			end
+			if t > 0 && t < tmin
+				tmin = t
+			end
+		end
+	end
+	if tmin <= 1
+		return tmin
+	else
+		return [0]
+	end
+end
+
+# returns the next t to compute the next weight vector w(t) = w + t * (tw - w) like it´s done in Cox, Little & O'Sheao (2005).
+function next_weight_fractal_walk(
+	G::Oscar.IdealGens,
+	cweight::Array{T, 1},
+	tweight::Array{K, 1},
```

Comment:
```suggestion
	cweight::Vector{T},
	tweight::Vector{K},
```
Any other use of `Array` should likely be removed, too